assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js

```diff
@@ -85,6 +85,9 @@ export default function SetupBanner( { onSubmitSuccess } ) {
 		select( MODULES_ANALYTICS ).getAccounts()
 	);
 
+	const getPropertyID = useSelect(
+		( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID
+	);
```

Comment:
This selects the actual selector function which is not something we have done much (or at all before?) in components, but is ok to do particularly in event handlers where we want to reference the current value at that moment.

In this case, we need it to avoid triggering dependency changes unnecessarily in a few places below where the ID itself was referenced before. The ID changes during the submit changes process (particularly when creating a new property) which in turn triggers the effect which "depends" on it to run again (unnecessarily).

Reply:
Interesting solution, something to bear in mind for future reference!

Reply:
@aaemnnosttv @techanvil this is anti-pattern IMO. We already have an approach for the same problem in other components that relies on a state variable. See here: https://github.com/google/site-kit-wp/blob/cff533d2b423176517f1e1f719cc3288be441627/assets/js/modules/adsense/components/setup/SetupMain.js#L274-L307

Reply:
Hmm, that is a valid concern @eugene-manuilov. It's worth noting that the docs for [`useSelect`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useselect) do discourage return the selector function from `useSelect`, stating:

> Don’t use useSelect this way when calling the selectors in the render
function because your component won’t re-render on a data change.

That said, it's less complicated than keeping track of an extra state variable. I'd be inclined to say let's keep it, but add a comment to make it clear that the non-standard usage is deliberate.

Reply:
I think consistency is more important than complications added by an extra state variable. I don't remember where it was mentioned but a while ago we had a statement in our engineering workflow (or in another document) that we encouraged using consistent and easy to understand approaches over smart ones that nobody would expect to see.

Reply:
Consistency is important, that's true. Personally, as mentioned I think we could keep it with a comment but I certainly have no issue with changing it either.